### PR TITLE
Initial review

### PR DESCRIPTION
**Pull request by sergiodj**
Hey @lucaskanashiro,

This is the initial review of the package.  As I said earlier, there's
not much to comment.  I think it's a good idea to start with pretty
much the same package we already have in the archive, and then
progressively improve it later.

Let me know when you address the comments and I can take a second
look.

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -129,6 +129,8 @@ override_dh_shlibdeps:
 override_dh_auto_clean:
 	@# stop debhelper from doing "make clean"
 
+### sergiodj: I tried invoking dh_dwz passing --no-dwz-multifile to
+### see if it could make any difference, to no avail.  Oh, well...
```
Comment (lucaskanashiro) [inline debian/rules:133]: I tried the same :) do you think we should improve the comment message here?

File: debian/docker.io.postinst [hunk 2]
```diff
@@ -15,6 +15,10 @@ case "$1" in
 		# Ensure config directory permissions
 		#  On a fresh install, $2 = '' and dpkg "le-nl" treat that as a
 		#  greater version, so the body is not executed.
+                ### sergiodj: Maybe we can take this opportunity to
+                ### clean this up?  Version 1.11.2~ds1-1 doesn't seem
+                ### to be supported even in Bionic, which is almost
+                ### EOLing.
```
Comment (lucaskanashiro) [inline debian/docker.io.postinst:21]: Right, I think we can do it indeed. I did not pay too much attention to those cleanup opportunities to try to not increase the diff of what we have in the archive, but I think this one is really straightforward. 

+1 for this change.

File: debian/docker.io.config [hunk 3]
```diff
@@ -1,5 +1,43 @@
...
+### sure that the question only gets asked once, but as far as I could
+### find there's no way to know whether the config script is being
+### executed during dpkg-preconfigure or during the postinst phase (if
+### there was, then I guess debconf would probably use it instead of
+### implementing its own deduplication mechanism).  So, in the
+### interest of making progress here, my suggestion is to revert this
+### change and put this code back into the postint script, ideally
+### with a comment on top explaining why it is there and not in a
+### config script.  You can also override the lintian warning, as we
+### discussed previously.
```
Comment (lucaskanashiro) [inline debian/docker.io.config:39]: Thank you for the detailed explanation @sergiodj ! Sure, let's move this code back to postinst, add a comment explaining why it is not in config, and override the lintian warning.

File: debian/control [hunk 4]
```diff
@@ -47,8 +47,6 @@ Suggests: aufs-tools,
           docker-doc,
           rinse,
           zfs-fuse | zfsutils
-Conflicts: docker.io (<< 20.10.25-0ubuntu1~)
-Replaces: docker.io (<< 20.10.25-0ubuntu1~)
```
Comment (lucaskanashiro) [inline debian/control:51]: Thank you for testing this :) +1

File: debian/copyright [hunk 5]
```diff
@@ -1,4 +1,4 @@
-Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
```
Comment (lucaskanashiro) [inline debian/copyright:1]: opss...

**Review by lucaskanashiro — COMMENTED** [review]
Thank you for the review @sergiodj !

File: debian/rules (hunk 1 above)
Comment (sergiodj) [reply debian/rules:133]: Yes, that's a good idea.

**Comment by sergiodj** [comment]
Awesome, +1 from me :-).

**Comment by sergiodj** [comment]
Lucas updated his branch with the following commit, which addresses my comments:

https://github.com/canonical/docker.io-app/commit/3e6e69d8518c8953ce6b69a841fb35ef960eab64

Therefore, I'm closing this PR.

**Pull request closed without merging**